app/models/custom_button_set.rb

```diff
@@ -1,6 +1,19 @@
 class CustomButtonSet < ApplicationRecord
   acts_as_miq_set
 
+  before_save :update_children
+  def update_children
+    if set_data.nil? || set_data[:button_order].nil?
+      remove_all_children
+      return
+    end
+
+    children = Rbac.filtered(CustomButton.where(:id => set_data[:button_order]))
+    throw(:abort) if children.pluck(:id).sort != set_data[:button_order].sort
+
+    replace_children Rbac.filtered(CustomButton.where(:id => set_data[:button_order]))
```

Comment:
Maybe it's wrong, but can't we use `find` here, then you don't need the `:id` at all and AFAIK it fails with invalid IDs.

Reply:
I changed that to `where` after feedback from @lpichler .